File: 31-60/maxSubArray.js

function maxSubArray(nums) {
    let max_value = -Infinity;
    let current_value = 0;

    for (let num of nums) {
        current_value = Math.max(num, current_value + num);
        max_value = Math.max(max_value, current_value);
    }
    return max_value;
}

const nums = [-2,1,-3,4,-1,2,1,-5,4];
console.log(maxSubArray(nums));

File: 31-60/maxSubArray.java
public class maxSubArray {
    public static int maxSubArr(int[] nums) {
        int maxSum = Integer.MIN_VALUE;
        int currentValue = 0;
        for (int num: nums) {
            currentValue += num;
            maxSum = Math.max(maxSum, currentValue);
            if (currentValue < 0) {
                currentValue = 0;
            }
        }
        return maxSum;
    }


    public static void main(String[] args) {
        int[] nums ={-2,1,-3,4,-1,2,1,-5,4};
        System.out.println(maxSubArr(nums));
    }
 
}
